tpot/tpot.py

```diff
@@ -61,6 +63,10 @@ class Bool(object):
     """Boolean class used for deap due to deap's poor handling of ints and booleans"""
     pass
 
+class MaskableInput(list):
+    def __getitem__(self, index):
+        try: return super(MaskableInput, self).__getitem__(index)
+        except TypeError: return MaskableInput(compress(self, index))
```

Comment:
IMO let's drop the MaskableInput and just use pandas DataFrame indexing for now. `.loc` should be decently fast for masking.